prov/util/src/util_buf.c

```diff
@@ -106,18 +130,34 @@ void *util_buf_get(struct util_buf_pool *pool)
 
 	entry = slist_remove_head(&pool->buf_list);
 	buf = container_of(entry, union util_buf, entry);
+	buf_ftr = (struct util_buf_footer *)
+		((char *) buf + pool->data_sz);
+
+	if (context)
+		*context = buf_ftr->region->context;
 
 #if ENABLE_DEBUG
-	pool->num_used++;
+	buf_ftr->region->num_used++;
 #endif
-	return buf;
+	return buf->data;
+}
+
+void *util_buf_get(struct util_buf_pool *pool)
+{
+	return util_buf_ext_get(pool, NULL);
 }
```

Comment:
I would reverse this implementation in order to avoid the overhead with util_buf_get.  Have util_buf_get_ex() or util_bufx_get() or whatever it's called call util_buf_get(), then handle the footer processing.